src/features/swap/helpers/utils.ts

```diff
@@ -10,7 +10,11 @@ function asDecimal(amount: number | bigint, decimals: number): number {
   return Number(formatUnits(amount, decimals))
 }
 
-export const getExecutionPrice = (order: SwapOrder): number => {
+type SwapOrderProps = Pick<
+  SwapOrder,
+  'executedBuyAmount' | 'executedSellAmount' | 'buyToken' | 'sellToken' | 'sellAmount' | 'buyAmount' | 'kind'
+>
```

Comment:
I pass the whole order because I'm lazy.

The order in the history is of type SwapOrder. The order in the confirmation View is of type CowSwapConfirmationView. I think that we kind of screwed the definition of those things on the server. I don't understand why they need to be of different type when they basically have the same props.

So back to your question - the SwapOrder type is "Swaporder", where the CowSwapconfirmationView is COW_SWAP_ORDER and as such I would have to modify the label functions to either accept a swapOrde or a CowSwapConfirmationView, but in fact the utility functions just care about a subset of values that's why I opted for extracting a new type with just the props we care about.